### ISSUE TITLE
File system suddenly "missing" after update

### ISSUE THREAD

**Issue by Xedon420** [openmediavault/openmediavault#143]
Hi,

I updated to the latest OMV version a few days ago and since then I have a big problem..... (here a screenshot of the error: http://prntscr.com/k9a45r)
Here is the sysinfo: http://prntscr.com/k9a4iv

At the beginning I didn't notice the error because everything went on just like before, only when I logged into the panel again yesterday and wanted to assign a folder to a user did I notice the error. All files are undamaged, installed services like plex, smb etc continue to work on the file system. Afterwards I made an ad bug responsible, so I restarted once but unfortunately the same error again... File system is still intact and can be used without problems so I conclude there is some error in the panel. 

I would be happy about help ;)

![fstab](https://user-images.githubusercontent.com/1897962/43133963-af6c070a-8f40-11e8-8aff-d699ec24ee5f.png)
![info](https://user-images.githubusercontent.com/1897962/43133964-af965622-8f40-11e8-8d5f-9495d764bad7.png)

**Comment by votdev** [comment]
The filesystem is not mounted, so please press the „Mount“ button.

**Comment by Xedon420** [comment]
I can't press it :(
the button is grayed out

**Comment by votdev** [comment]
WHat is the state of the RAID containing the filesystem?

# cat /proc/mdstat

**Comment by Xedon420** [comment]
active: http://prntscr.com/k9th5n

**Comment by votdev** [comment]
What happens if you mount the filesystem manually via CLI?

P.S.: You can add images here. Images using external services are not available anymore some day, thus this bugreport becomes useless. The best is to copy/paste the raw text.

**Comment by Xedon420** [comment]
The Filesystems is mounted... Samba, plex can use it seems for my only as a display bug..

**Comment by votdev** [comment]
What is the output of ``blkid -o full``?

**Comment by Xedon420** [comment]
``root@NAS:~# blkid -o full
/dev/sda: UUID="2174d5ce-5e03-4234-f6c4-80edd024cf9a" UUID_SUB="3915b4a3-ba3e-8692-c622-c18ff7491675" LABEL="NAS:Storage" TYPE="linux_raid_member"
/dev/sdb: UUID="2174d5ce-5e03-4234-f6c4-80edd024cf9a" UUID_SUB="9f477511-f301-8b02-df97-579e25528d56" LABEL="NAS:Storage" TYPE="linux_raid_member"
/dev/sde1: UUID="3ed73ad8-279b-43da-b7cd-bdb26c89244a" TYPE="ext4" PARTUUID="b1fc64cd-01"
/dev/sde5: UUID="c07f6696-a3ac-4188-ac9e-4bc0f113b1e1" TYPE="swap" PARTUUID="b1fc64cd-05"

**Comment by votdev** [comment]
There is the problem, you have two file systems with the same label. Remove one of the label to fix this issue.

**Comment by Xedon420** [comment]
I haven't added another label and I can't accept a loss of data... I can't explain it otherwise (I can't delete from a file system at the moment). Is there another solution than the webpanel?

**Comment by Xedon420** [comment]
what if you can do this without losing data? How?

**Comment by votdev** [comment]
I tried to reproduce this issue, but wasn't able. Everything works. What OMV version are you using?

Please also post the output of ``mount``.

**Comment by Xedon420** [comment]
Ich verstehs nicht...
Alle Dienste können die Festplatte wie immer nutzen nur das OMV Webpanel hat plötzlich nach einem Update ein Fehler... Ich meine wenn ein zweites label dafür verantwortlich ist muss dieses zweite label ja irgendwie erstellt worden sein. 

OMV Version: 4.1.8.2-1 Arrakis is installed

Hier die Ausgabe vom Command mount...
`root@NAS:~# mount
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
udev on /dev type devtmpfs (rw,nosuid,relatime,size=1977220k,nr_inodes=494305,mode=755)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,noexec,relatime,size=405060k,mode=755)
/dev/sde1 on / type ext4 (rw,relatime,errors=remount-ro,data=ordered)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/lib/systemd/systemd-cgroups-agent,name=systemd)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=40,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=10349)
mqueue on /dev/mqueue type mqueue (rw,relatime)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
sunrpc on /run/rpc_pipefs type rpc_pipefs (rw,relatime)
tmpfs on /tmp type tmpfs (rw,relatime)
nfsd on /proc/fs/nfsd type nfsd (rw,relatime)
/dev/md0 on /srv/dev-disk-by-id-md-name-NAS-Storage type ext4 (rw,noexec,relatime,data=ordered,jqfmt=vfsv0,usrjquota=aquota.user,grpjquota=aquota.group)
tmpfs on /run/user/997 type tmpfs (rw,nosuid,nodev,relatime,size=405056k,mode=700,uid=997,gid=996)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,relatime)
tmpfs on /run/user/1005 type tmpfs (rw,nosuid,nodev,relatime,size=405056k,mode=700,uid=1005,gid=1001)`

**Comment by votdev** [comment]
What is the output of ``omv-confdbadm read --prettify "conf.system.filesystem.mountpoint"``?

**Comment by Xedon420** [comment]
`root@NAS:~# omv-confdbadm read --prettify "conf.system.filesystem.mountpoint"
[
    {
        "dir": "/srv/dev-disk-by-id-md-name-NAS-Storage",
        "freq": 0,
        "fsname": "/dev/disk/by-id/md-name-NAS:Storage",
        "hidden": false,
        "opts": "defaults,nofail,user_xattr,noexec,usrjquota=aquota.user,grpjquota=aquota.group,jqfmt=vfsv0,acl",
        "passno": 2,
        "type": "ext4",
        "uuid": "c3aa31c9-53b6-4da7-915b-4cab8550a63d"
    }
]
`

**Comment by votdev** [comment]
Still no idea what is going wrong. What is the output of ``udevadm info --query=property --name=/dev/md0``?

**Comment by votdev** [comment]
I think i've found the problem, but i don't know the reason for this. On your system there is no blkid entry for the /dev/md0 device like on my system:

```
root@omv4box:/home/vagrant# blkid
/dev/vda1: UUID="199a4bbc-59c9-4a3b-b592-950eaffb2530" TYPE="ext4" PARTUUID="1aac37af-01"
/dev/sdb: UUID="f376f1dd-2fb1-e1a7-5b66-eac09348e857" UUID_SUB="93522a42-7d44-3de6-a46d-8388acfc094c" LABEL="omv4box:Storage" TYPE="linux_raid_member"
/dev/md127: UUID="00015783-6d7e-41b3-a58d-f3b5292f3c1b" TYPE="ext4"
/dev/sdc: UUID="f376f1dd-2fb1-e1a7-5b66-eac09348e857" UUID_SUB="f7136555-6410-4616-490e-653e1bb114f7" LABEL="omv4box:Storage" TYPE="linux_raid_member"
/dev/sda1: PARTUUID="5d35e9e5-a1aa-4714-8e26-c09cdefc59b1"
```

**Comment by Xedon420** [comment]
hmm ok...
An idea how to fix this?

**Comment by votdev** [comment]
I'm sorry, no. This problem is far out of the scope of OMV. Seems to be a problem with the kernel <-> md <-> filesystem.

**Comment by votdev** [comment]
You did not answer how this MD RAID was created. A solution is to reinstall the Debian/OMV version where the filesystem on your MD device is accessible to create a backup of your data. Then setup a complete system with OMV4 to workaround this Debian <-> kernel <-> md module <-> filesystem issue.

**Comment by Xedon420** [comment]
hmm ok...

I cannot perform a new installation at the moment. (I have no more backup disks free)
Is there a workaround how I can set the users without the panel etc?

But it's a bit strange because it really worked until the last update without problems and now suddenly it doesn't work anymore... By the way, I use a pure OMV image here.

**Comment by Xedon420** [comment]
ok means, as soon as a new kernel update comes up, it could be that the error resolves itself again or?

**Comment by votdev** [comment]
This could be, but i wouldn't bet on it.

**Comment by brando56894** [comment]
I have the same issue. I created my RAID10 array on Arch Linux and imported it in OMV, it _did_ show up but during the process of updating and installing things, it disappeared. 

Here's all the info you requested above: https://hastebin.com/tesaviniku.makefile

**Comment by votdev** [comment]
@brando56894 Thanks for the report. I think this is a bug in the kernel or userland, maybe something related to the MD metadata (i'm no expert in this case). Sadly i have to say this is out of the scope of the OMV project.

**Comment by brando56894** [comment]
Interestingly enough....it's back now! I noticed while doing the above that my array wasn't mounted at /mnt/storage like it usually is, I so changed the mount point, but that screwed up a lot of things OMV related since it was expecting it to be mounted under /srv, so I mounted the array at both places, rebooted...and now it exists about 4 hours after a reboot...but it wasn't there initially after the reboot. Very strange. 

I'm going to reinstall OMV since I bought an NVMe drive, and if I have time I'll recreate the array in OMV this time. Thanks.
